bundler/lib/bundler.rb

```diff
@@ -353,7 +353,9 @@ def unbundled_env
       env.delete_if {|k, _| k[0, 7] == "BUNDLE_" }
 
       if env.key?("RUBYOPT")
-        env["RUBYOPT"] = env["RUBYOPT"].sub "-rbundler/setup", ""
+        rubyopt = env["RUBYOPT"].split(" ")
+        rubyopt.delete("-r#{File.expand_path("bundler/setup", __dir__)}")
+        env["RUBYOPT"] = rubyopt.join(" ")
```

Comment:
I followed the pattern in RUBYLIB below and split the string into an Array on " ", then deleted the `bundler/setup` entry, then joined the RUBYOPT array back together with " ". I did that to avoid having to deal with white space when deleting the entry.

For example, if RUBYOPT contained`-r/path/to/bundler/setup -r/something/else` and I just subbed out `-r/path/to/bundler/setup` for `""` RUBYOPT would now have a leading space left over.

But, if I tried to remove `-r/path/to/bundler/setup ` (with trailing space) then it wouldn't work when it was the ONLY entry because there would be no trailing space. It would also not work if it was the last entry.

Splitting on " " helps us avoid all of that because it will "just work" for any of the above cases and it won't leave the leading space behind in RUBYOPT.

The default for `String#split` is " " so I don't technically need to specify it, but I went ahead and specified it explicitly here for clarity (so that it's clearer what is happening)